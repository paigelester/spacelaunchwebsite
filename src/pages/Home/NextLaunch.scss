@import 'App.scss';

.next-launch {
    position: relative;

    &-image-container {
        width: 100%;
        max-height: 90vh;
        overflow: hidden;

        img {
            max-width: 100%;
            height: auto;
        }
    }

    &-details {
        position: absolute;
        width: 100%;
        background-color: $brand-panel-colour;

        .badge {
            color: $black;
            font-weight: normal;
            background-color: $brand-yellow;
        }

        p {
            color: $sub-text-color;
        }

        
    }
}