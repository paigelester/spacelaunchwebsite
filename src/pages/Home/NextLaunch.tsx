import React from "react";
import { Container, Row, Col, Image, Card } from "react-bootstrap";

import LaunchResource from 'api/LaunchResource';

import Countdown from "components/features/Countdown";
import LaunchStatusIndicator from "components/features/LaunchStatusIndicator";

const NextLaunch = (props: LaunchResource) => {
    return (
        <div className='shadow-lg'>
            <Container fluid>
                <Row>
                    <Col md={12} lg={9} className="g-0">
                        <div className="position-relative overflow-hidden">
                            <Image src={props.image} alt='launch' className="img-fluid" />
                            <div className="position-absolute top-0 w-100 h-100 o-75 bg-dark"></div>
                            <div className="position-absolute top-50 start-50 end-0 translate-middle text-center w-100">
                                <p className="text-uppercase text-center mb-1 fs-2">Time to launch</p>
                                <Countdown targetDate={props.window_start} />
                            </div>
                        </div>
                    </Col>
                    <Col className="g-0">
                        <Card className="h-100">
                            <Card.Body>
                                <div className="float-start">
                                    <p className="d-inline fs-6 text-muted border-end border-dark pe-2 me-2">{props.launch_service_provider?.name}</p>
                                    <p className="d-inline fs-6 text-muted">{props.launch_service_provider?.type}</p>
                                </div>
                                {props.status && <div className="float-end"><LaunchStatusIndicator status={props.status} /></div>}
                                <h4 className="pt-5 pb-2">{props.name}</h4>
                                <p>{props.mission?.description}</p>
                            </Card.Body>
                        </Card>
                    </Col>
                </Row>
            </Container>
        </div>
    );
};

export default NextLaunch;