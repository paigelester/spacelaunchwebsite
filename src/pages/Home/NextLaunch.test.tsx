import React from "react";
import { render } from "@testing-library/react";
import NextLaunch from "./NextLaunch";

beforeEach(() => {
    Date.now = jest.fn(() => 1618171860892);
});

it("renders as expected", () => {
    // given
    const name: string = "Launch name";
    const status: string = "Go for Launch";
    const launchProviderName: string = "Launch Co";
    const launchProviderType: string = "Commercial";
    const launchWindowStart: string = new Date(Date.now()).toString();
    const missionDescription: string = "This is an example description of the launch mission.";

    // when
    const { container } = render(
        <NextLaunch
            name={name}
            image={""}
            status={status}
            launchWindowStart={launchWindowStart}
            missionDescription={missionDescription} 
            launchProviderName={launchProviderName}
            launchProviderType={launchProviderType} />
    );

    // then
    expect(container).toMatchSnapshot();
});
