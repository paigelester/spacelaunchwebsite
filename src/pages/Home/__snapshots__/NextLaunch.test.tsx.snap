// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="shadow-lg"
  >
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="g-0 col-lg-9 col-md-12"
        >
          <div
            class="position-relative overflow-hidden"
          >
            <img
              alt="launch"
              class="img-fluid"
              src=""
            />
            <div
              class="position-absolute top-0 w-100 h-100 o-75 bg-dark"
            />
            <div
              class="position-absolute top-50 start-50 end-0 translate-middle text-center w-100"
            >
              <p
                class="text-uppercase text-center mb-1 fs-2"
              >
                Time to launch
              </p>
              <div
                class="countdown"
              >
                <div
                  class="text-center me-2 d-inline-block"
                >
                  <p
                    class="countdown-days-value fs-1 m-0"
                  >
                    1
                  </p>
                  <p
                    class="countdown-days text-muted text-uppercase m-0"
                  >
                    days
                  </p>
                </div>
                <div
                  class="text-center me-2 d-inline-block"
                >
                  <p
                    class="countdown-hours-value fs-1 m-0"
                  >
                    00
                  </p>
                  <p
                    class="countdown-hours text-muted text-uppercase m-0"
                  >
                    hours
                  </p>
                </div>
                <div
                  class="text-center me-2 d-inline-block"
                >
                  <p
                    class="countdown-minutes-value fs-1 m-0"
                  >
                    59
                  </p>
                  <p
                    class="countdown-minutes text-muted text-uppercase m-0"
                  >
                    minutes
                  </p>
                </div>
                <div
                  class="text-center d-inline-block"
                >
                  <p
                    class="countdown-seconds-value fs-1 m-0"
                  >
                    59
                  </p>
                  <p
                    class="countdown-seconds text-muted text-uppercase m-0"
                  >
                    seconds
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="g-0 col"
        >
          <div
            class="h-100 card"
          >
            <div
              class="card-body"
            >
              <div
                class="float-start"
              >
                <p
                  class="d-inline fs-6 text-muted border-end border-dark pe-2 me-2"
                >
                  Launch Co
                </p>
                <p
                  class="d-inline fs-6 text-muted"
                >
                  Commercial
                </p>
              </div>
              <div
                class="float-end"
              >
                <span
                  class="badge bg-success"
                >
                  Go for Launch
                </span>
              </div>
              <h4
                class="pt-5 pb-2"
              >
                Launch name
              </h4>
              <p>
                This is an example description of the launch mission.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
